app/src/main/java/app/grapheneos/camera/capturer/VideoCapturer.kt

```diff
@@ -52,48 +52,33 @@ class VideoCapturer(private val mActivity: MainActivity) {
             if (isRecording) {
                 if (value) {
                     recording?.pause()
-                    pauseTimer()
                     mActivity.flipCamIcon.setImageResource(R.drawable.play)
                 } else {
                     recording?.resume()
-                    startTimer()
                     mActivity.flipCamIcon.setImageResource(R.drawable.pause)
                 }
             }
             field = value
         }
 
     private val handler = Handler(Looper.getMainLooper())
-    private var elapsedSeconds: Int = 0
-    private val runnable = Runnable {
-        ++elapsedSeconds
-        val secs = padTo2(elapsedSeconds % 60)
-        val min = padTo2(elapsedSeconds / 60 % 60)
-        val hours = padTo2(elapsedSeconds / 3600)
+
+    private fun padTo2(time: Long): String {
+        return String.format("%1$" + 2 + "s", time).replace(' ', '0')
+    }
+
+    private fun updateTimerTime(timeInNanos: Long) {
+        val timeInSec = timeInNanos / (1000 * 1000 * 1000)
+        val secs = padTo2(timeInSec % 60)
+        val min = padTo2(timeInSec / 60 % 60)
+        val hours = padTo2(timeInSec / 3600)
         val timerText: String = if (hours == "00") {
             "$min:$secs"
         } else {
             "$hours:$min:$secs"
```

Comment:
`String.format(Locale.ROOT, "%02d:%02d:%02d", hours, min, sec)` would be more understandable